### DISCUSSION TITLE
Are functions deployed in a way that hasura actions can easily access them?

### DISCUSSION THREAD

**Discussion by wollerman** [nhost/nhost#1752] [Q&A] [answered]
I'm looking to use the nhost functions and connect hasura actions. I can put the FQDN in the `handler` field here:
<img width="370" alt="image" src="https://user-images.githubusercontent.com/1610241/226139199-501f1ab4-f15f-48bf-b6c3-86bcdb4ed8f7.png">

But I was wondering if there was a shorter path than having to have hasura make a request to a public endpoint?

Perhaps another way to frame the question: are functions and hasura available to eachother to avoid ingress/egress? It would be awesome if they happen to be on the same internal network, but I don't have much insight into the deployment setup.

Relatedly, because hasura is a huge piece of nhost, is there any documentation, best practices, or recomendations on setting up a hasura action with nhost functions?

Thanks!

**Comment by dbarrosop** [comment, marked as answer]
Hasura and functions run on different networks but they are in the same AWS region so even though you are using a public URL your traffic doesn't need to traverse the internet.

> on setting up a hasura action with nhost functions?

Docs can be found here: [event triggers](https://docs.nhost.io/database/event-triggers)

**Reply by wollerman** [reply]
This is great, thanks! I was looking under the `serverless functions` section as I wasn't necessarily working on using database event triggers for this piece. It might make sense to move that documentation to the `serverless function` section and reference it. I can create a PR if you'd like.

Thanks again!

**Reply by dbarrosop** [reply]
No problem. Rather than moving it entirely maybe add a reference. I am afraid if we move it someone will complain it isn't in the other place and will want to move it again :P

**Reply by wollerman** [reply]
Haha fair enough! Here's a PR that just adds a quick link so users know to look there https://github.com/nhost/nhost/pull/1772
